### PR TITLE
refactor(#1904): `PlaceMojo#place` method. Decrease complexity.

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
I've decreased complexity of `PlaceMojo#place` method by introducing additional inner class `BinaryDependency`.

Closes: #1935

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
Blocked by #1933

**Comment by volodya-lombrozo** [comment]
@mximp Reminder

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java [hunk 1]
```diff
@@ -119,16 +119,23 @@ private static Stream<String> stream(final Collection<String> globs) {
      * @return List
      * @throws IOException If fails
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java:120]: @volodya-lombrozo doesn't throw any more

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public final class Walk extends ListEnvelope<Path> {
      * @param dir The directory
      * @throws IOException If fails
```
Comment (mximp) [inline eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java:54]: @volodya-lombrozo doesn't throw

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java]: @mximp Indeed. Removed. Thank you.

File: eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java (hunk 2 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/util/Walk.java]: @mximp Indeed. Removed. Thank you.

File: eo-maven-plugin/src/main/java/org/eolang/maven/PlaceMojo.java [hunk 3]
```diff
@@ -90,22 +90,16 @@ public final class PlaceMojo extends SafeMojo {
      * @since 0.30
      * @checkstyle MemberNameCheck (7 lines)
      */
-    private final PlacedTojosCache placedTojosCache = new PlacedTojosCache(this.placedTojos);
+    private PlacedTojosCache placedCache = new PlacedTojosCache(this.placedTojos);
```
[the remark this answers is not in the document]
Comment (mximp) [reply eo-maven-plugin/src/main/java/org/eolang/maven/PlaceMojo.java:93]: not critical

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you have a look, please?

**Comment by volodya-lombrozo** [comment]
@yegor256 Reminder

**Comment by yegor256** [comment]
@rultor merge

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you try to merge it again, please?

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor stop

**Comment by yegor256** [comment]
@rultor merge

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you just revert the changes, please? It blocks development.

**Comment by yegor256** [comment]
@rultor stop

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor stop

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**